File: C#/7_Binary Search/Binary Search.cs
using System;

namespace Algorithm {

    class BinarySearch {

        public static int binarySearch(int[] arr, int l, int r, int x) {
            if(r >= l) {
                int mid = l + (r - l) / 2;

                if(arr[mid] == x){
                    return mid;
                }

                if(arr[mid] > x){
                    return binarySearch(arr, l, mid - 1, x);
                } else {
                    return binarySearch(arr, mid + 1, r, x);
                }
            }
            return -1;
        }

        public static void Main(string[] args){
            int[] arr = { 3, 4, 23, 42, 77, 100 };
            int x = 42;
            Console.Write(binarySearch(arr, 0, arr.Length - 1, x));
        }
    }
}

File: Swift/7_Binary Search/Binary Search.swift
func binarySearch(arr: [Int], value: Int) -> Int? {
    var array = arr
	var left = 0
    var right = array.count - 1

    while left <= right {

        let middle = Int(Double(left + right) / 2.0)

        if array[middle] < value {
            left = middle + 1
        } else if array[middle] > value {
            right = middle - 1
        } else {
            return middle
        }
    }
    return nil
}

let array = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
let x = 13

if let index = binarySearch(arr: array, value: x) {
    print("Found \(x) at index \(index)")
} else {
    print("Did not find \(x)")
}